hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+Once the command has started, the core will assert {{< regref "STATUS.ACTIVE" >}}.
+The command is complete when {{< regref "STATUS.ACTIVE" >}} goes low.
+An `event` interrupt can also be triggered to go off on completion by setting {{< regref "EVENT_ENABLE.CMD_CMPLT" >}}.
+
+**TODO**: Describe TX1, TXN, dummy, RXN etc.  Refer to previous waveform.
+
+### Chip Select ID's
+
+Each instance of the SPI_HOST IP support up to 8 chip select lines (`cs_n[MaxCS-1:0]`).
+Each `cs_n` line can be routed either to a single peripheral or to a daisy-chain of peripherals.
+The three bit {{< regref "COMMAND.CSID" >}} field in each command indicates that the line `cs_n[COMMAND.CSID]` should be asserted for that command.
+The SPI_HOST core typically then manages the details of asserting and deasserting the proper `cs_n` line, subject to the timing parameters expressed in {{< regref "CONFIG.CSAAT" >}}, {{< regref "CONFIG.CSNLEAD" >}}, {{< regref "CONFIG.CSNTRAIL" >}}, and {{< regref "CONFIG.CSNIDLE" >}}.
+
+Alternatively the `cs_n` lines can be controlled entirely by firmware.
+If the register {{< regref "CONTROL.MANCS_EN" >}} is set high then the `cs_n` lines are directly controlled by the {{< regref "CONTROL.MANUAL_CS" >}} register.
+
+If [Pass-through mode](#pass-through-mode) is enabled then the `cs_n` lines are controlled by *neither* the SPI_HOST hardware nor the firmware register.
+In Pass-though mode, control of the `cs_n` lines passes directly to the inter-module port, `pt_cs_ni`.
+
+### Arbitrary Length Commands
+
+Typically the length of each commands are limited by the maximum number of bits in the {{< regref "COMMAND.TXN_CNT" >}}, {{< regref "COMMAND.RX_CNT" >}}, and {{< regref "COMMAND.TX1_CNT" >}} fields.
+A pair of SPI_HOST IP commands however can be merged into one long command sequence using the {{< regref "CONFIG.CSAAT" >}} field, which leaves the `cs_n` line low after the first SPI_HOST command.
+By raising the `cs_n` line only after the second command has completed, the peripheral will interpret the resulting transmit/receive sequences as one long command.
+
+This method can be also applied to a series of SPI_HOST IP commands, to create commands of arbitrary length.
+
+### Opportunistic Support for Back-to-back Commands
+
+There is no command queue in the SPI_HOST IP, however a second command can be placed into the Config/Command CDC whenever {{< regref "STATUS.READY" >}} is high, even if the previous command is still running.
+The internal state within the Config/Command CDC provides the option of scheduling a second command to execute immediately after completion of the current command.
+
+On the other hand, writing a one to {{< regref "COMMAND.GO" >}} when {{< regref "STATUS.READY" >}} is low will trigger an error condition, and must be acknowledged by software.
```

Comment:
one approach we've taken in several modules is that once "go" is set, the registers are locked from updates until the transaction is complete.

Reply:
Yeah, as mentioned in my comment above, that seems like it could give software false confidence.  Is there an error that goes back from the TL-UL block?

As long as firmware has *some* indication that something went wrong, I'm happy to adapt that method.

Reply:
This comment is to be cited in #5157, and will hopefully be addressed in next PR, #5149.

Reply:
To be addressed in issue